velox/functions/prestosql/MapSubset.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/ComplexViewTypes.h"
+#include "velox/functions/Udf.h"
+
+namespace facebook::velox::functions {
+
+/// Fast path for constant primitive type keys: map_subset(m, array[1, 2, 3]).
+#define VELOX_MAP_SUBSET_PRIMITIVE_FUNCTION(TName, TType)           \
```

Comment:
why macro and not template like ArrayJoinFunction

Reply:
@laithsakka Laith, thank you for the pointer. I didn't know about ParameterBinder. Let me see if I can get rid of the macro. Somehow, I didn't see ParameterBinder described in the docs.

```
  registerFunction<
      ParameterBinder<ArrayJoinFunction, T>,
      Varchar,
      Array<T>,
      Varchar>({prefix + "array_join"});
```

Reply:
@laithsakka Laith, thanks again or the tip. I was above to remove the macro and use ParameterBinder : #9216